bucket_registry.go

```diff
@@ -56,7 +65,9 @@ func (r *bucketRegistry) getCachedBucket(name string) *Bucket {
 	if bucket == nil {
 		return nil
 	}
-	return bucket.copy()
+	// return a copy of the bucket
+	_, bucket = r._registerBucket(bucket)
```

Comment:
It seems redundant to call _registerBucket here which is going to extract the name and do the same lookup in r.buckets that we just did.  I feel like it would be better to just do this here:
```
r.bucketCount[name] += 1
return bucket.copy()
```
and then move the _registerBucket into registerBucket.

Reply:
Fixed.